Library/Homebrew/formula_specialties.rb

```diff
@@ -18,12 +18,18 @@ def self.url(val)
 # This formula serves as the base class for several very similar
 # formulae for Amazon Web Services related tools.
 class AmazonWebServicesFormula < Formula
+
   # Use this method to peform a standard install for Java-based tools,
   # keeping the .jars out of HOMEBREW_PREFIX/lib
   def install
     rm Dir['bin/*.cmd'] # Remove Windows versions
     libexec.install Dir['*']
-    bin.install_symlink Dir["#{libexec}/bin/*"] - ["#{libexec}/bin/service"]
+    Pathname.glob("#{libexec}/bin/*") do |file|
+      next if file.directory?
+      basename = file.basename
+      next if basename.to_s == "service"
+      (bin/basename).write_env_script file, :JAVA_HOME => ENV["JAVA_HOME"]
```

Comment:
Like this:
```ruby
module Langugae
  module Java
    def self.java_home_env(version=nil)
      { :JAVA_HOME => "$(/usr/libexec/java_home#{version ? " -v #{version}" : ""})" }
    end
  end
end
```

Also I kinda want to move JavaDependency into a separate file(like PythonDependency etc.) Is this OK?